lib/repo/git_repository.rb

```diff
@@ -66,8 +73,62 @@ def self.create(connect_string)
                           some directory with same name exists already")
       end
 
-      #Create it (we're not going to use a bare repository)
-      repo = Rugged::Repository.init_at(connect_string)
+      ga_repo = Gitolite::GitoliteAdmin.new(
+        Repository.conf[:REPOSITORY_STORAGE] +
+          '/gitolite-admin', GITOLITE_SETTINGS)
+
+      # Bring the repo up to date
+      ga_repo.reload!
+
+      # Grab the gitolite admin repo config
+      conf = ga_repo.config
+
+      repo_name = File.basename(connect_string.split('/').last)
+
+      # Grab the repo in question, if it does not exist, create it
+      repo = ga_repo.config.get_repo(repo_name)
+      if repo.nil?
+          # Generate new repo since this repo hasn't been created yet
+          repo = Gitolite::Config::Repo.new(repo_name)
+      end
+
+      # Add permissions for git user
+      repo.add_permission('RW+', '', 'git')
+
+      # Add the repo to the gitolite admin config
+      conf.add_repo(repo)
+
+      # Readd the 'git' public key to the gitolite admin repo after changes
+      self.class.readd_admin_key
+
+      # Stage and push the changes to the gitolite admin repo
+      ga_repo.save_and_apply
+
+      # Repo is created by gitolite, proceed to clone it in
+      # the repository storage location
+      cloned_repo = Git.clone(
+        'git@localhost:' + repo_name, REPOSITORY_STORAGE + '/' + repo_name)
+
+      repo = Rugged::Repository.discover(REPOSITORY_STORAGE + '/' + repo_name)
+
+      # Lets make some sample files and the new master branch
+      cloned_repo.reset
+      cloned_repo.branch('master')
+
+      # Create the README.md on the filesystem
+      file_path_for_readme = File.join(repo.workdir, 'README.md')
+      File.open(file_path_for_readme, 'w+') do |readme|
+        readme.write('Initial commit.')
+      end
+
+      cloned_repo.add(:all=>true)
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Surrounding space missing for operator '=>'.